framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInference.java

```diff
@@ -120,17 +122,21 @@ void updateInferredFieldType(
 
     /**
      * Updates the return type of the method {@code methodTree} based on {@code returnedExpression}.
+     * May also strengthen the return types of any abstract methods that this method overrides.
```

Comment:
This weakens them rather than strengthens them, I believe.  I believe that all changes made by WPI are inital assignment of a qualifier, or weakenings (LUBs).  (Is that necessary/desirable?  Why not propagate in the other direction as well and do GLBs?

Or do you mean compared to the default qualifier?  Even then, the initial assignment could be a weakening or a strengthening.

Reply:
I had this same question when re-reading the diffs.